### ISSUE TITLE
Incorrect value for "ghinja.ghidra_install_path" in ~/Library/Application Support/Binary Ninja/settings.json Causes Crash on Launch

### ISSUE THREAD

**Issue by leaky-hub** [Vector35/binaryninja-api#5864]
**Version and Platform (required):**
- Binary Ninja Version: 4.1.5902 Personal, f2165c5d
- OS: macos
- OS Version: 14.6
- CPU Architecture: arm64

**Bug Description:**
Prerequisite: Ghinja plugin is enabled. 
The ghinja plugin requires a valid value / path for "ghinja.ghidra_install_path" in settings.json in order for the application to launch correctly. Users that move the location of Ghidra and do not update the value / path will encounter a crash on launch.

**Steps To Reproduce:**
1. Install the Ghinja plugin and have Ghidra installed / downloaded
2. Close Binary Ninja
3. After entering the initial location of analyzeHeadless.bat, move the Ghidra directory (e.g. from downloads to documents).
4. Launch Binary Ninja
5. Observe the warning to enter location of analyzeHeadless.bat:
![image](https://github.com/user-attachments/assets/49cbebc3-1a22-4aaf-a519-5ff4023c45e4)
6. Hit 'OK' and observe the "Binary Ninja quit unexpectedly." error:
![image](https://github.com/user-attachments/assets/f6bedede-048c-45f9-8e09-f6d885daaec3)

**Expected Behavior:**
Users should be able to enter the new location of "analyzeHeadless.bat" with validation and / or if location is invalid, Binary Ninja should launch without crashing so the user can re-enter the correct path in the plugin settings.

**Additional Information**
Users can fix this by going into ~/Library/Application Support/Binary Ninja/settings.json and readjusting the value of "ghinja.ghidra_install_path" to the correct one:
![image](https://github.com/user-attachments/assets/c0596051-188f-4bca-b7b4-0857409f0755)

**Comment by xusheng6** [comment]
This is an issue in https://github.com/Martyx00/ghinja, which is third-party plugin that we do not control

**Comment by plafosse** [comment]
Please file an issue on the ghinja plugin repository. We will look into the crash as we shouldn't be crashing regardless.

**Comment by fuzyll** [comment]
This is the line of code in the plugin in question that is causing the crash: https://github.com/Martyx00/ghinja/blob/fe3fab3d7e51fe53cd6a75279465c08413043bc3/__init__.py#L31

The plugin itself won't install and run anymore because it uses the now-removed `DockHandler`. I've tried to re-create the issue with a mashed-together version of the code that matters and haven't been able to. I'm guessing the crash either has something to do with the way the message box was created or some bug that used to exist in either Binary Ninja or Qt.

Without an actual crash report, and without the plugin being actively maintained (or working at all, as of next release), I am going to close this as not planned. If anyone comes across another situation that crashes like this and has more information we can use to reproduce it, please reopen this or make a new issue and we'll take a look.
